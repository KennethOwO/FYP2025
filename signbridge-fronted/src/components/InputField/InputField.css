.form-group {
    margin-bottom: 50px;
    position: relative;
    max-height: 33%;
}

.form-group input,
.form-group textarea {
    width: 100%;
    height: 50px;
    outline: 0;
    padding: 10px 10px 10px 16px;
    border: 0px solid #000000;
    border-radius: 4px;
    position: relative;
    font-size: 17px;
    background-color: rgb(231, 231, 231);
}

.form-group .input-label {
    position: absolute;
    top: 50%;
    left: 3%;
    transform: translateY(-50%);
    font-size: 18px;
    color: #666;
    transition: 0.2s all;
    text-align: left;
    cursor: text;
    pointer-events: none;
}

.input-label.multiple_label {
    transform: translateY(-180%);
}

.form-group input:focus + .input-label,
.form-group input:not(:placeholder-shown) + .input-label {
    top: -15px;
    font-size: 18px;
    left: 0.5%;
    color: #000;
}

.form-group textarea:focus ~ .input-label,
.form-group textarea:not(:placeholder-shown) ~ .input-label {
    top: 10px;
    font-size: 18px;
    left: 0.5%;
    color: #000;
}

.form-group input:not(:placeholder-shown),
.form-group textarea:not(:placeholder-shown) {
    background-color: white;
    border: 1px solid #000;
}

.form-group input:focus,
.form-group textarea:focus {
    border: #000 2px solid;
    background-color: white;
}

.form-group textarea {
    height: auto;
    min-height: 100px;
    resize: none;
    overflow: auto;
}

/* .form-control.is-invalid {
    border-color: red !important;
}

.form-group .invalid-feedback {
    color: red;
    font-size: 15px;
    margin-left: 10px;
    text-align: left;
}

.form-group.is-invalid input:focus+.input-label,
.form-group.is-invalid input:placeholder-shown+.input-label {
    margin-top: -10px;
} */

.form-group .form-error {
    opacity: 0;
    position: absolute;
    transition: opacity 0.3s ease-in-out;
    margin-left: 5px;
    font-size: 12px;
    color: red;
}

.form-group .form-error.is-invalid {
    opacity: 1;
}

.form-control.is-invalid {
    border-color: red !important;
}

@media screen and (max-width: 768px) {
    .form-group input,
    .form-group textarea {
        padding: 8px;
    }
}

@media screen and (max-width: 480px) {
    .form-group input,
    .form-group textarea {
        padding: 6px;
    }
}
