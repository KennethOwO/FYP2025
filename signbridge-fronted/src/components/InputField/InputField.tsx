import React, { ChangeEvent } from "react";
import "./InputField.css";

interface InputFieldProps {
  label: string;
  name: string;
  placeholder?: string;
  type?: string;
  value?: string;
  defaultValue?: string;
  id?: string;
  onChange: (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
  multipleLines?: boolean;
  error: string;
}

const InputField: React.FC<InputFieldProps> = ({
  label,
  name,
  placeholder = " ",
  type = "text",
  value = "",
  defaultValue,
  onChange,
  error,
  id,
  multipleLines = false,
}) => {
  const inputElement = multipleLines ? (
    <textarea
      className={`form-control ${error ? "is-invalid" : ""} multipleLines`}
      placeholder={placeholder}
      name={name}
      value={value}
      id={id}
      onChange={onChange}
      data-testid={name}
      defaultValue={defaultValue}
    />
  ) : (
    <input
      className={`form-control ${error ? "is-invalid" : ""}`}
      placeholder={placeholder}
      type={type}
      name={name}
      value={value}
      id={id}
      onChange={onChange}
      data-testid={name}
      defaultValue={defaultValue}
    />
  );

  return (
    <div className={`form-group ${error ? "is-invalid" : ""}`}>
      {inputElement}
      <label className={`input-label ${multipleLines ? "multiple_label" : ""}`}>
        {label}
      </label>
      <div className={`form-error ${error ? "is-invalid" : ""}`}>{error}</div>
    </div>
  );
};

export default InputField;
