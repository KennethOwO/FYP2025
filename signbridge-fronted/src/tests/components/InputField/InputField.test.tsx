import React from "react";
import InputField from "@root/components/InputField/InputField";
import { create } from "react-test-renderer";

describe("Test InputField", () => {
  it("should render correctly", () => {
    const props = {
      label: "Test Label",
      name: "test",
      onChange: jest.fn(), // Mocking the onChange function
      error: "Test Error Message",
    };
    const tree = create(<InputField {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
