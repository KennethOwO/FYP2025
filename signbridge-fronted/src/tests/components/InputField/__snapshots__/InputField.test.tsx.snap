// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputField should render correctly 1`] = `
<div
  className="form-group is-invalid"
>
  <input
    className="form-control is-invalid"
    data-testid="test"
    name="test"
    onChange={[MockFunction]}
    placeholder=" "
    type="text"
    value=""
  />
  <label
    className="input-label "
  >
    Test Label
  </label>
  <div
    className="form-error is-invalid"
  >
    Test Error Message
  </div>
</div>
`;
